core/src/io/anuke/mindustry/input/InputHandler.java

```diff
@@ -418,17 +419,50 @@ protected void flushRequests(Array<BuildRequest> requests){
                 if(copy.hasConfig && copy.block.posConfig){
                     copy.config = Pos.get(Pos.x(copy.config) + copy.x - copy.originalX, Pos.y(copy.config) + copy.y - copy.originalY);
                 }
+
+                copy.block = didyoumean(req.x, req.y, req.block, req.rotation);
+
                 player.addBuildRequest(copy);
             }
         }
     }
 
+    protected Block didyoumean(int x, int y, Block block, int rotation){
```

Comment:
I would call this `checkReplacement` or something similar. Additionally, it would be a good idea to make replacements get defined *in the blocks themselves*, so that each block could return a block it could be replaced with given a build queue:

```java
//in Block.java
public Block getReplacement(int x, int y, int rotation, Array<BuildRequest> requests /* or Eachable<BuildRequest>*/){
    return this; //override to return a junction, etc
}
```

This would allow blocks like conduits to override this behavior easily and implement replacements for liquid junctions.